pxtblocks/blocklycompiler.ts

```diff
@@ -679,6 +679,18 @@ namespace pxt.blocks {
         return H.mathCall(op, args);
     }
 
+    function compileFunctionDefinition(e: Environment, b: Blockly.Block, comments: string[]): JsNode[] {
+        const name = escapeVarName(b.getFieldValue("function_name"), e, true);
+        const stmts = getInputTargetBlock(b, "STACK");
+        const argsDeclaration = (b as Blockly.FunctionDefinitionBlock).getArguments().map(a => {
+            return `${escapeVarName(a.name, e)}: ${a.type}`;
```

Comment:
Escape how?

Reply:
You need to TS-escape the name if it's coming from untrusted XML, because someone could have maliciously changed it. TypeScript has two functions call `isIdentifierStart()` and `isIdentifierPart()` for removing illegal characters.